Fixes to JSONMonotoring (74X)

smorovic opened the pull request:
- fix calculating checksum on a wrong buffer (cause of spurious crashes)
- suppression of output when FastMonitoringService value is reset to 0

smorovic committed:
*fix stat where reading from name buffer instead of content buffer when calculating Adler32
*suppression of output when FastMonitoringService value is reset to 0

cmsbuild commented:
A new Pull Request was created by @smorovic (Srecko Morovic) for CMSSW_7_4_X.

Fixes to JSONMonotoring (74X)

It involves the following packages:

HLTrigger/JSONMonitoring


@Martin-Grunewald, @perrotta, @cmsbuild, @fwyzard can you please review it and eventually sign? Thanks.
@Martin-Grunewald this is something you requested to watch as well.
You can sign-off by replying to this message having '+1' in the first line of your reply.
You can reject by replying  to this message having '-1' in the first line of your reply.
If you are a L2 or a release manager you can ask for tests by saying 'please test' in the first line of a comment.
@degano you are the release manager for this.
You can merge this pull request by typing 'merge' in the first line of your comment.

smorovic committed:
using bool as a flag

cmsbuild commented:
Pull request #9613 was updated. @Martin-Grunewald, @perrotta, @cmsbuild, @fwyzard can you please check and sign again.

fwyzard commented on HLTrigger/JSONMonitoring/plugins/TriggerJSONMonitoring.cc:
```diff
@@ -573,6 +579,14 @@ TriggerJSONMonitoring::globalEndLuminosityBlockSummary(const edm::LuminosityBloc
     outHltJsnData.close();
 
+    //HLT jsn entries
+    StringJ hltJsnFilelist_;
```
hi Srecko,
please do not use names ending with an underscore for non-member variables.

.A

fwyzard commented on HLTrigger/JSONMonitoring/plugins/TriggerJSONMonitoring.cc:
```diff
@@ -529,7 +531,14 @@ TriggerJSONMonitoring::endLuminosityBlockSummary(const edm::LuminosityBlock& iLu
+  unsigned int iLs  = iLumi.luminosityBlock();
+  unsigned int iRun = iLumi.run();
+
+  evf::FastMonitoringService * fms_ = (evf::FastMonitoringService *)(edm::Service<evf::MicroStateService>().operator->());
```
Please add a check for the availability of the FastMonitoringService or MicroStateService.
It is fine to print a LogError and/or disable some functionality if a service is not available, but it should not crash the whole program.

(actually, IMO *nothing* should ever crash the whole program, if a more graceful alternative is available)

fwyzard commented on HLTrigger/JSONMonitoring/plugins/TriggerJSONMonitoring.cc:
```diff
@@ -529,7 +531,14 @@ TriggerJSONMonitoring::endLuminosityBlockSummary(const edm::LuminosityBlock& iLu
+  unsigned int iRun = iLumi.run();
+
+  evf::FastMonitoringService * fms_ = (evf::FastMonitoringService *)(edm::Service<evf::MicroStateService>().operator->());
+  bool writeFiles = fms_ ? fms_->getEventsProcessedForLumi(iLs)>0 : true;
```
Ops, apologies, ==fms_== is used only here, and there is a check - fooled by the name, I though it was a member available elsewhere.

smorovic committed:
* removing trailing '_' from local variables
* check if MicroStaterService is available for better readability

smorovic commented:
@fwyzard 
I changed all local variables to not use "_" (although this is Aram's code, I'm just fixing a bug...).
There is a check for the service, but I changed it a bit to be more readable.

cmsbuild commented:
Pull request #9613 was updated. @Martin-Grunewald, @perrotta, @cmsbuild, @fwyzard can you please check and sign again.

Martin-Grunewald commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

fwyzard commented:
hi @smorovic , I was looking only at the diff, but thanks for renaming the variables elsewhere, too.

cmsbuild commented:
+1
Tested at: d196a8c5bd7babd9c7a79fa0eb6b428117395906
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-9613/5481/summary.html

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-9613/5481/summary.html

Martin-Grunewald commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_7_4_X IBs once checked with relvals in the development release cycle of CMSSW (tests are also fine). This pull request requires discussion in the ORP meeting before it's merged. @davidlange6, @degano, @smuzaffar

davidlange6 commented:
+1
